kapp_platforms/src/windows/event_loop_windows.rs

```diff
@@ -36,6 +36,41 @@ pub unsafe extern "system" fn window_callback(
         }
         WM_KEYDOWN | WM_SYSKEYDOWN => produce_event(process_key_down(w_param, l_param)),
         WM_KEYUP | WM_SYSKEYUP => produce_event(process_key_up(w_param, l_param)),
+        WM_CHAR => {
+            let character = String::from_utf16(&[w_param as u16])
+                .unwrap()
+                .chars()
+                .next()
+                .unwrap();
+            produce_event(Event::CharacterReceived { character });
+        }
+        WM_IME_STARTCOMPOSITION => {
+            return TRUE as isize;
+        }
+        WM_IME_ENDCOMPOSITION => {
+            produce_event(Event::IMEEndComposition);
+            return TRUE as isize;
+        }
+        WM_IME_COMPOSITION => {
+            let himc = ImmGetContext(hwnd);
+            if himc == null_mut() {
+                return 0;
+            }
+            if l_param as u32 & GCS_COMPSTR != 0 {
+                let len = ImmGetCompositionStringW(himc, GCS_COMPSTR, null_mut(), 0);
+                if len == 0 {
+                    return 0;
+                }
+                let mut buffer = Vec::<u16>::with_capacity(len as usize / 2);
```

Comment:
This is a nitpick, but I'd prefer `/ std::mem::size_of::<u16>()` here because it makes more explicit what's going on.